src/Utils/ASTDefinitionBuilder.php

```diff
@@ -329,7 +329,10 @@ private function getDeprecationReason($node)
         return $deprecated['reason'] ?? null;
     }
 
-    private function makeImplementedInterfaces(ObjectTypeDefinitionNode $def)
+    /**
+     * @param ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode $def
+     */
+    private function makeImplementedInterfaces($def)
```

Comment:
What does this return?

Reply:
This either returns `null` or an array `Type` instances (which is what `$this->buildType` officially returns). If we assume a correct implementation of `$this->buildType` then we expect the return to be `?array<int,InterfaceType>`

Reply:
Given we have a comprehensive test suite, you can just put in the proper type hints and we will see if they hold true. If not, that is even better and gives us a chance to clarify or fix a bug.

Reply:
```suggestion
    private function makeImplementedInterfaces($def) : ?array
```